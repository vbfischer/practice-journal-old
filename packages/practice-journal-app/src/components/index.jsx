import MenuButton from './molecules/MenuButton/MenuButton';
import Profile from './molecules/Profile/Profile';
import MainTemplate from './templates/MainTemplate/MainTemplate';

import DashboardPage from './pages/DashboardPage/DashboardPage';

import Header from './organisms/Header/Header';
import Sidebar from './organisms/Sidebar/Sidebar';
import Article from './organisms/Article/Article';

import SidebarItem, { NavSidebarItem } from './atoms/SidebarItem/SidebarItem';
import Staff from './atoms/Tabulature/Staff';
import theme from './themes';

export {
  MainTemplate,
  DashboardPage,
  Header,
  theme,
  MenuButton,
  Sidebar,
  SidebarItem,
  NavSidebarItem,
  Profile,
  Article,
  Staff
};
