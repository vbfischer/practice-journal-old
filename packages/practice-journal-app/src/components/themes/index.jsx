import { createMuiTheme } from '@material-ui/core/styles';

const theme = createMuiTheme({
  typography: {
    useNextVariants: true
  },
  palette: {
    primary: {
      main: '#f4f3ef'
    },
    secondary: {
      main: '#66615B'
    }
  }
});

export default theme;
