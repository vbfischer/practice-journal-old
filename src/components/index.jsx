import MenuButton from './molecules/MenuButton/MenuButton';

import MainTemplate from './templates/MainTemplate/MainTemplate';

import DashboardPage from './pages/DashboardPage/DashboardPage';

import Header from './organisms/Header/Header';
import Sidebar from './organisms/Sidebar/Sidebar';

import SidebarItem, { NavSidebarItem } from './atoms/SidebarItem/SidebarItem';
import theme from './themes';

export {
  MainTemplate,
  DashboardPage,
  Header,
  theme,
  MenuButton,
  Sidebar,
  SidebarItem,
  NavSidebarItem
};
