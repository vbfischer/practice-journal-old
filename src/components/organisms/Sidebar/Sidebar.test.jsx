import React from 'react';

import { shallow, mount } from 'enzyme';

import Button from '@material-ui/core/Button';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import List from '@material-ui/core/List';
import { Sidebar } from '../../../components';

describe('<Sidebar/>', () => {
  describe('render title', () => {
    const title = 'The Title';
    const component = shallow(<Sidebar title={title} />);

    it('should match snapshot', () => {
      expect(component).toMatchSnapshot();
    });

    it('should set property title', () => {
      expect(component.instance().props.title).toBe(title);
      expect(
        component
          .dive()
          .find(Button)
          .children()
          .text()
      ).toBe(title);
    });
  });

  describe('render header', () => {
    const heading = <div>Heading...</div>;

    const component = shallow(<Sidebar heading={heading} />);

    it('should match snapshot', () => {
      expect(component).toMatchSnapshot();
    });

    it('should set property heading', () => {
      expect(component.instance().props.heading).toBe(heading);
    });

    it('should insert heading into dom', () => {
      expect(component.dive().containsMatchingElement(heading)).toBeTruthy();
    });
  });

  describe('render children', () => {
    const Children = () => (
      <React.Fragment>
        <ListItem>
          <ListItemText primary="something" />
        </ListItem>
      </React.Fragment>
    );
    const component = shallow(
      <Sidebar>
        <Children />
      </Sidebar>
    );

    it('should match snapshot', () => {
      expect(component).toMatchSnapshot();
    });

    it('should insert children', () => {
      expect(component.dive().find(Children).length).toBe(1);
    });
  });
});
