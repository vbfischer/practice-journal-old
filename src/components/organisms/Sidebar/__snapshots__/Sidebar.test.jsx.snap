// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar/> render children should match snapshot 1`] = `
<Sidebar
  classes={
    Object {
      "drawerPaper": "Sidebar-drawerPaper-1",
      "drawerPaperClose": "Sidebar-drawerPaperClose-2",
      "list": "Sidebar-list-6",
      "logo": "Sidebar-logo-3",
      "logoLink": "Sidebar-logoLink-4",
      "toolbar": "Sidebar-toolbar-5",
    }
  }
>
  <Children />
</Sidebar>
`;

exports[`<Sidebar/> render header should match snapshot 1`] = `
<Sidebar
  classes={
    Object {
      "drawerPaper": "Sidebar-drawerPaper-1",
      "drawerPaperClose": "Sidebar-drawerPaperClose-2",
      "list": "Sidebar-list-6",
      "logo": "Sidebar-logo-3",
      "logoLink": "Sidebar-logoLink-4",
      "toolbar": "Sidebar-toolbar-5",
    }
  }
  heading={
    <div>
      Heading...
    </div>
  }
/>
`;

exports[`<Sidebar/> render title should match snapshot 1`] = `
<Sidebar
  classes={
    Object {
      "drawerPaper": "Sidebar-drawerPaper-1",
      "drawerPaperClose": "Sidebar-drawerPaperClose-2",
      "list": "Sidebar-list-6",
      "logo": "Sidebar-logo-3",
      "logoLink": "Sidebar-logoLink-4",
      "toolbar": "Sidebar-toolbar-5",
    }
  }
  title="The Title"
/>
`;
